prow/plugins/plugins.go

```diff
@@ -279,13 +382,58 @@ func (pa *PluginAgent) PushEventHandlers(owner, repo string) map[string]PushEven
 	return hs
 }
 
+// Config returns the plugin configuration for the repo, which
+// will either be the owning organization's config, the repo
+// config or a merged version of the two
+func (pa *PluginAgent) PluginConfig(owner, repo string) Config {
+	pa.mut.Lock()
+	defer pa.mut.Unlock()
+
+	fullName := fmt.Sprintf("%s/%s", owner, repo)
+	orgConfig, orgOk := pa.pc[owner]
+	repoConfig, repoOk := pa.pc[fullName]
+
+	if repoOk && !orgOk {
+		return repoConfig
+	} else if !repoOk && orgOk {
+		return orgConfig
+	} else if orgOk && repoOk {
+		return mergeConfig(orgConfig, repoConfig)
+	}
+
+	panic(fmt.Sprintf("No configuration was found for %s!", fullName))
+}
+
+// mergeConfig merges organization and repository plugin config
+// to create the full configuration for a set of repository plugins.
+func mergeConfig(org, repo Config) Config {
```

Comment:
Instead, we could resolve configuration at the start if we know the set of repos we will be looking at